<launch>
    <!-- px4_device param applies a frame offset (for the 'droneX_origin' frame to work around px4 origin limitations -->
    <arg name="px4_device" default="true"/> 
    <!-- Drone id -->
    <arg name="drone_id"/>
    <arg name="drone_id_ns" value="drone$(arg drone_id)"/>
    <arg name="num_drones"/>
    <arg name="init_x"/>
    <arg name="init_y"/>
    <arg name="init_z"/>
    <!-- size of map, change the size inflate x, y, z according to your application -->
    <arg name="map_size_x"/>
    <arg name="map_size_y"/>
    <arg name="map_size_z"/>
    <!-- Input/output topics -->
    <arg name="command_topic"/>
    <arg name="odometry_topic"/>
    <arg name="pose_topic"/>
    <arg name="depth_img_topic"/>
    <arg name="point_cloud_topic"/>
    <!-- number of moving objects -->
    <arg name="obj_num" value="10" />
    <!-- drone_origin_frame: For PX4 SITL, leave it as default. 
    For other forms of simulation, might have to change it -->
    <arg name="drone_origin_frame" default="$(arg drone_id_ns)_origin" />

    <!-- Trajectory server -->
    <node pkg="trajectory_server" name="drone_$(arg drone_id)_traj_server" type="traj_server_node" output="screen">
        <param name="drone_id" value="$(arg drone_id)"/>
        <param name="origin_frame" value="$(arg drone_origin_frame)"/>
        <!-- <param name="drone_model_mesh" value="file://$(find gestelt_bringup)/simulation/meshes/fake_drone.dae"/> -->
        <param name="drone_model_mesh" value="file:///home/john/gestelt_ws/src/gestelt/gestelt_bringup/simulation/meshes/fake_drone.dae"/>

        <!-- Trajectory params -->
        <param name="traj_server/time_forward" value="1.0" type="double"/>
        <param name="traj_server/max_poses_to_track" value="1000" type="int"/>

        <!-- Frequency for timers -->
        <param name="traj_server/pub_cmd_freq" value="25.0" type="double"/>
        <param name="traj_server/state_machine_tick_freq" value="50.0" type="double"/>
        <param name="traj_server/takeoff_height" value="1.0" type="double"/>

        <!-- Subscribed Topics -->
        <remap from="~planning/trajectory" to="/drone_$(arg drone_id)_planning/trajectory"/>
        <remap from="/planner_state" to="/$(arg drone_id_ns)/planner_state"/>
        <remap from="/mavros/state" to="/$(arg drone_id_ns)/mavros/state"/>
        <remap from="/mavros/local_position/pose" to="/$(arg drone_id_ns)/$(arg pose_topic)"/>
        <remap from="/mavros/local_position/odom" to="/$(arg drone_id_ns)/$(arg odometry_topic)"/>
        <!-- Published Topics -->
        <remap from="/mavros/setpoint_raw/local" to="/$(arg drone_id_ns)/$(arg command_topic)"/>
        <remap from="/server_state" to="/$(arg drone_id_ns)/server_state"/>
        <remap from="/uav_path" to="/$(arg drone_id_ns)/uav_path"/>
        <remap from="/tracking_error" to="/$(arg drone_id_ns)/tracking_error"/>
        <remap from="~model" to="/$(arg drone_id_ns)/model"/>
        <!-- Service clients -->
        <remap from="/mavros/cmd/arming" to="/$(arg drone_id_ns)/mavros/cmd/arming"/>
        <remap from="/mavros/set_mode" to="/$(arg drone_id_ns)/mavros/set_mode"/>
    </node>

</launch>
